htsinfer/infer_read_layout.py

```diff
@@ -1,8 +1,276 @@
 """Infer read layout from sample data."""
+from collections import Counter
+from typing import List
+
+import numpy as np
 
 
 def infer():
     """Main function coordinating the execution of all other functions.
     Should be imported/called from main app and return results to it.
     """
     # implement me
+
+
+def randomize_nucleotide_sequence(
+        input_sequences: List,
+        number_random_seq: int = 1,
+        pseudo_count: int = 1,
+        n_freq: float = 0.05
+        ) -> List:
+    """
+    Returns randomised sequences adhering to the dinucleotide probabilites
+    found in the input sequences.
+
+    Args:
+        input_sequences: List of input sequences as strings.
+        number_random_seq: Number of randomised sequences returned per input
+            sequence.
+        pseudo_count: Pseudo count for each dinucleotide before the input
+            sequences are counted. This ensures compatibility with input
+            sequences that do not contain all nucleotide combinations. Can
+            be set to 0 if all combinations occur in the input. Can be used
+            to weigh the probabilites of the input seuqences versus random
+            probabilites.
+
+    Returns:
+        number_random_seq x randomised sequences of equal length to the input
+        sequence in a list.
+
+    Raises:
+        ValueError: input_string contains too many characters other than ATGCN.
+        ValueError: The percentage of N's one of the input sequences is higher
+            than n_freq.
+        ValueError: Input sequence is empty.
+        ValueError: Input sequence is not of type 'str'
+        ValueError: number_random_seq < 1.
+        ValueError: pseudo_count < 0.
+    """
+
+    # Error if input sequences list is empty
+    if len(input_sequences) == 0:
+        raise ValueError("List of input sequences is empty")
+
+    input_seqs_upper = []
+    for i, sequence in enumerate(input_sequences):
+        # Error if any of the input is not a str
+        if not isinstance(sequence, str):
+            raise ValueError(
+                f"Input sequence {i} is not of type 'str'\n"
+                f"{sequence}"
+            )
+
+        # Convert input strings to upper case
+        sequence = sequence.upper()
+        input_seqs_upper.append(sequence)
+
+        # Error if empty sequence is entered
+        if len(sequence) == 0:
+            raise ValueError(
+                f"Sequence {i} is empty"
+            )
+        # Count the occurences of N's
+        count = Counter(sequence)
+        if count["N"] >= n_freq * len(sequence):
```

Comment:
This test is too strict. There should at least be a flag that allows the program to continue if some of the sequences in the input file are not valid (because they contain too many N's or they contain characters that are not [A, C, G, T, N]